### DISCUSSION TITLE
multiple wslconfig

### DISCUSSION THREAD

**Discussion by souradeep100** [microsoft/WSL#6912] [Q&A]
Hi All, is it possible to have multiple wslconfig file for different distributions?
Also is wsl2 still uses single kernel for all the distributions? Or we can have different kernel images, for different distributions?
Also as all pCPU are visible as vCPU to wsl2, what percentage per CPU is allowed?

**Comment by Biswa96** [comment]
`/etc/wsl.conf` is already a thing.

**Reply by souradeep100** [reply]
User can not set their CPU and Memory for the distro in /etc/wsl.conf, as that file comes into picture for kernel during boot up, but
wslconfig file comes for hyper-v from root partition which is windows10 in this case. 
So though u can set some mount and dns options in wsl.conf but you can not set the hyper-v specific things there.

**Reply by Biswa96** [reply]
vCPU and vMemory are taken into action before booting the VM. So, the `/etc/wsl.conf` will not considered that time.

**Reply by souradeep100** [reply]
Yes that I have mentioned already. So if we need per distro based vCPU and vMem, like we can do from normal hyper-v for per VM, or any hypervisor cli, here we don't have that provision. But that will be helpful, if we can segregate that in .wslconfig file,
with per vm name/ id.

**Reply by Biswa96** [reply]
Hmm... That's a challenge 🤔 There is only one VM running in WSL2. With your case, there will be multiple.

**Reply by souradeep100** [reply]
Now if we want to run one kali linux and one ubuntu simultaneously, then we
will need that.
That's why I have asked for multiple kernel images.

**Reply by Biswa96** [reply]
No, the challenge is more bigger than using multiple Linux kernel. Currently, if one runs one Kali and one Ubuntu in WSL2 those two only uses one VM. If there are multiple VM there will be multiple VM ID. WSL2 relies on vSockets heavily and vSockets are associated with each VM ID. The COM interfaces are also linked with vSockets. And many more...
